contracts/WildcardSteward_v3.sol

```diff
@@ -353,13 +353,39 @@ contract WildcardSteward_v3 is Initializable {
     function changeReceivingBenefactor(
         uint256 tokenId,
         address payable _newReceivingBenefactor
-    ) public onlyReceivingBenefactorOrAdmin(tokenId) {
+    )
+        public
+        onlyReceivingBenefactorOrAdmin(tokenId)
+        updateBenefactorBalance(benefactors[tokenId])
+        updateBenefactorBalance(_newReceivingBenefactor)
+    {
         address oldBenfactor = benefactors[tokenId];
-        require(
-            oldBenfactor != _newReceivingBenefactor,
-            "Cannot change to same address"
-        );
+
+        require(oldBenfactor != _newReceivingBenefactor);
+        require(address(0) != _newReceivingBenefactor);
+
+        // Collect patronage from old and new benefactor before changing totalBenefactorTokenNumerator on both
+        uint256 scaledPrice = price[tokenId].mul(patronageNumerator[tokenId]);
+        totalBenefactorTokenNumerator[oldBenfactor] = totalBenefactorTokenNumerator[oldBenfactor]
+            .sub(scaledPrice);
+        totalBenefactorTokenNumerator[_newReceivingBenefactor] = totalBenefactorTokenNumerator[_newReceivingBenefactor]
+            .add(scaledPrice);
```

Comment:
Here we make the adjustments for the totalBenefactorNumerator for the old and new. We can do this since we just updated the balance of both.